services/galley/src/Galley/API/Internal.hs

```diff
@@ -691,28 +691,30 @@ rmUser lusr conn = do
       let qUser = qUntagged lusr
       cc <- getConversations ids
       now <- input
-      pp <- for cc $ \c -> case Data.convType c of
-        SelfConv -> pure Nothing
-        One2OneConv -> deleteMembers (Data.convId c) (UserList [tUnqualified lusr] []) $> Nothing
-        ConnectConv -> deleteMembers (Data.convId c) (UserList [tUnqualified lusr] []) $> Nothing
-        RegularConv
-          | tUnqualified lusr `isMember` Data.convLocalMembers c -> do
+      let deleteIfNeeded c =
+            when (tUnqualified lusr `isMember` Data.convLocalMembers c) $ do
               runError (removeUser (qualifyAs lusr c) (qUntagged lusr)) >>= \case
                 Left e -> P.err $ Log.msg ("failed to send remove proposal: " <> internalErrorDescription e)
                 Right _ -> pure ()
               deleteMembers (Data.convId c) (UserList [tUnqualified lusr] [])
-              let e =
-                    Event
-                      (qUntagged (qualifyAs lusr (Data.convId c)))
-                      (qUntagged lusr)
-                      now
-                      (EdMembersLeave (QualifiedUserIdList [qUser]))
               for_ (bucketRemote (fmap rmId (Data.convRemoteMembers c))) $ notifyRemoteMembers now qUser (Data.convId c)
-              pure $
-                Intra.newPushLocal ListComplete (tUnqualified lusr) (Intra.ConvEvent e) (Intra.recipient <$> Data.convLocalMembers c)
-                  <&> set Intra.pushConn conn
-                    . set Intra.pushRoute Intra.RouteDirect
-          | otherwise -> pure Nothing
+          fireEvent c =
+            let e =
+                  Event
+                    (qUntagged (qualifyAs lusr (Data.convId c)))
+                    (qUntagged lusr)
+                    now
+                    (EdMembersLeave (QualifiedUserIdList [qUser]))
+             in pure $
+                  Intra.newPushLocal ListComplete (tUnqualified lusr) (Intra.ConvEvent e) (Intra.recipient <$> Data.convLocalMembers c)
+                    <&> set Intra.pushConn conn
+                      . set Intra.pushRoute Intra.RouteDirect
+      pp <- for cc $ \c -> case Data.convType c of
+        SelfConv -> pure Nothing
+        One2OneConv -> deleteMembers (Data.convId c) (UserList [tUnqualified lusr] []) $> Nothing
+        ConnectConv -> deleteMembers (Data.convId c) (UserList [tUnqualified lusr] []) $> Nothing
+        RegularConv -> deleteIfNeeded c >> fireEvent c
+        GlobalTeamConv -> deleteIfNeeded c >> pure Nothing
```

Comment:
Actually we need a different logic here. For the global team conv we need to do the MLS client removals regardless of membership, since that's implicit.